### PR TITLE
docs: add pybind11 example

### PR DESCRIPTION
**Pull request by alvgaona**
This is an example of using Pixi to compile pybind11 modules and make them available to any Python interpreter.

1. It uses CMake to compile the C++ code.
2. `scikit-build-core` is the tool used in the example to generate the `.whl` files.
3. It leverages the [build](https://pypi.org/project/build/) PyPI package to run the build task.

### CODE REVIEW

File: examples/pybind11/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+name = "mysum"
+version = "0.1.0"
+
+[tool.scikit-build]
+cmake.source-dir = "."
+wheel.packages = ["mysum"]
+
+[tool.pixi.project]
+channels = ["conda-forge"]
+platforms = ["linux-64"]
```
[the remark this answers is not in the document]
Comment (alvgaona) [reply examples/pybind11/pyproject.toml:16]: Yes, should work on those platforms as well.

**Comment by alvgaona** [comment]
> Thank you for your PR!
> 
> The build took 3 minutes on my machine, which feels a bit long for 10 lines of C++ code. I haven't used pybind11 before, so I don't know if that is normal.

@Hofer-Julian I'm not sure why that is. I think it's mainly  due to `scikit-build-core` doing things 🫠

**Comment by Hofer-Julian** [comment]
Maybe someone with more C++ experience like @baszalmstra or @wolfv could check it out as well

**Comment by alvgaona** [comment]
The composed command took ~6 seconds  when I built it with CMake directly. So I'm pretty sure it's `scikit-build-core`.

Screenshot 2024-10-21T15 42 44@2x

**Comment by alvgaona** [comment]
2m 33.07 seconds when running the full task on `osx-64`.
Screenshot 2024-10-21T15 46 43@2x

**Comment by alvgaona** [comment]
~17s with the `--wheel` flag.

Screenshot 2024-10-21T15 49 14@2x

**Comment by Hofer-Julian** [comment]
> ~17s with the `--wheel` flag. The task that takes too long is when it is ran with with the `--sdist` flag, which I think it's default.

Sounds to me like we should pass the `--wheel` flag then, since that's the interesting output in my opinion :)

**Comment by alvgaona** [comment]
> > ~17s with the `--wheel` flag. The task that takes too long is when it is ran with with the `--sdist` flag, which I think it's default.
> 
> Sounds to me like we should pass the `--wheel` flag then, since that's the interesting output in my opinion :)

Yeah, generally you'd like `--sdist` to share it via repositories. You can do whatever in this case.

**Comment by alvgaona** [comment]
@Hofer-Julian fixed it. The sdist was including `.pixi` in the tarballed package which was causing to take minutes, and by excluding it, it takes ~34s.

Screenshot 2024-10-22T03 08 01@2x

**Comment by beenje** [comment]
Hi! Just saw this PR. It's a nice example.

I wanted to mention that we also use `scikit-build-core` in one of our project (with `boost`).
I created an `install` task that runs `pip` to build and install the package in editable mode as defined [here](https://scikit-build-core.readthedocs.io/en/stable/configuration.html#editable-installs). It's still marked as experimental support but has been working well for us.

`pixi run install` builds and install the package.
It allows to work on the pure python part without re-installing. If you touch `cpp` files, you have to re-run the command.

You might find that useful.
I think it's more convenient during development than `pixi run build` + `pip install ...` that you have to re-run every time.
After it might depend on what you want and if your project is pure `cpp` or include python as well.

See https://gitlab.com/tango-controls/pytango/-/blob/develop/pixi.toml?ref_type=heads#L18

**Pull request merged**